BENCHMARK.md

```diff
@@ -0,0 +1,162 @@
+# Benchmark
+
+The main component of the **SDGym** project is the `sdgym.benchmark` function.
+
+As shown in the [README](README.md) quickstart, in the most simple usage scenario a synthesizer
+function can be directly passed to it:
+
+```python3
+from sdgym import benchmark
+
+scores = benchmark(my_synthesizer_function)
+```
+
+This will evaluate the synthesizer function on all the datasets, produce one or more scores
+for each one of them, and present them in a table alongside the scores previously obtained
+by all the SDGym Synthesizers:
+
+```
+                        adult/accuracy  adult/f1  ...  ring/test_likelihood
+IndependentSynthesizer         0.56530  0.134593  ...             -1.958888
+UniformSynthesizer             0.39695  0.273753  ...             -2.519416
+IdentitySynthesizer            0.82440  0.659250  ...             -1.705487
+...                                ...       ...  ...                   ...
+my_synthesizer_function        0.64865  0.210103  ...             -1.964966
+```
+
+However, the `benchmark` function supports a number of additional arguments that allows us
+to configure its behavior to better adapt it to our needs.
+
+## Benchmark Arguments
+
+The `benchmkark` function accepts the following arguments:
+
+* `synthesizers (function, class, list, tuple or dict)`:
+    The synthesizer or synthesizers to evaluate. It can be a single synthesizer
+    (function or method or class), or an iterable of synthesizers, or a dict
+    containing synthesizer names as keys and synthesizers as values. If the input
+    is not a dict, synthesizer names will be extracted from the given object.
+* `datasets (list[str])`:
+    Names of the datasets to use for the benchmark. Defaults to all the ones available.
+* `iterations (int)`:
+    Number of iterations to perform over each dataset and synthesizer. Defaults to 3.
+* `add_leaderboard (bool)`:
+    Whether to append the obtained scores to the previous leaderboard or not. Defaults
+    to `True`.
+* `leaderboard_path (str)`:
+    Path to where the leaderboard is stored. Defaults to the leaderboard included
+    with the package, which contains the scores obtained by the SDGym synthesizers.
```

Comment:
I would suggest not to overwrite the leaderboard inside the package. 

Basically, here are the expected behavior:

`add_leaderboard = False, leaderboard_path = None`: write new leader board to current directory.

`add_leaderboard = True, leaderboard_path = None`: write leader board to current directory, and append default leaderboard. 

`leaderboard_path is not None`: write leader board to a specified path. `add_leaderboard` decides overwriting or appending new results.

Reply:
We are not overwriting the leaderboard. This is the path where the current (or reference) leaderboard is stored, but it is never modified.
The new leaderboard is always returned as a DataFrame, and the user can then store it where they need it using `to_csv` on it.